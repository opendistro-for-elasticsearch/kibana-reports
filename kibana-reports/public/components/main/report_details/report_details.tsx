/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React, { useEffect, useState } from 'react';
import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiPage,
  EuiPageHeader,
  EuiTitle,
  EuiPageBody,
  EuiPageContent,
  EuiHorizontalRule,
  EuiSpacer,
  EuiDescriptionList,
  EuiDescriptionListTitle,
  EuiDescriptionListDescription,
  EuiPageHeaderSection,
  EuiButton,
  EuiLink,
  EuiIcon,
  EuiGlobalToastList,
} from '@elastic/eui';
import { fileFormatsUpper, generateReportById } from '../main_utils';
import { GenerateReportLoadingModal } from '../loading_modal';
import { ReportSchemaType } from '../../../../server/model';
import { converter } from '../../report_definitions/utils';
import dateMath from '@elastic/datemath';
import {
  permissionsMissingActions,
  permissionsMissingToast,
  timeRangeMatcher,
} from '../../utils/utils';

export const ReportDetailsComponent = (props) => {
  const { reportDetailsComponentTitle, reportDetailsComponentContent } = props;

  return (
    <EuiFlexItem>
      <EuiDescriptionList>
        <EuiDescriptionListTitle>
          {reportDetailsComponentTitle}
        </EuiDescriptionListTitle>
        <EuiDescriptionListDescription>
          {reportDetailsComponentContent}
        </EuiDescriptionListDescription>
      </EuiDescriptionList>
    </EuiFlexItem>
  );
};

// convert markdown to plain text, trim it if it's longer than 3 lines
export const trimAndRenderAsText = (markdown: string) => {
  if (!markdown) return markdown;
  const lines = markdown.split('\n').filter((line) => line);
  const elements = lines.slice(0, 3).map((line, i) => <p key={i}>{line}</p>);
  return lines.length <= 3 ? elements : elements.concat(<p key={3}>...</p>);
};

export const formatEmails = (emails: string[]) => {
  return Array.isArray(emails) ? emails.join(', ') : emails;
};

export function ReportDetails(props) {
  const [reportDetails, setReportDetails] = useState({});
  const [toasts, setToasts] = useState([]);
  const [showLoading, setShowLoading] = useState(false);

  const reportId = props.match['params']['reportId'];

  const handleLoading = (e) => {
    setShowLoading(e);
  };

  const addPermissionsMissingDownloadToastHandler = () => {
    const toast = permissionsMissingToast(
      permissionsMissingActions.GENERATING_REPORT
    );
    setToasts(toasts.concat(toast));
  };

  const handlePermissionsMissingDownloadToast = () => {
    addPermissionsMissingDownloadToastHandler();
  };

  const addErrorToastHandler = (title = 'Error loading report details.', text = '') => {
    const errorToast = {
      title,
      text,
      color: 'danger',
      iconType: 'alert',
      id: 'reportDetailsErrorToast',
    };
    setToasts(toasts.concat(errorToast));
  };

  const handleErrorToast = (title?: string, text?: string) => {
    addErrorToastHandler(title, text);
  };

  const addSuccessToastHandler = () => {
    const successToast = {
      title: 'Success',
      color: 'success',
      text: <p>Report successfully downloaded!</p>,
      id: 'onDemandDownloadSuccessToast',
    };
    setToasts(toasts.concat(successToast));
  };

  const handleSuccessToast = () => {
    addSuccessToastHandler();
  };

  const removeToast = (removedToast) => {
    setToasts(toasts.filter((toast) => toast.id !== removedToast.id));
  };

  const handleReportDetails = (e) => {
    setReportDetails(e);
  };

  const convertTimestamp = (timestamp: number) => {
    let displayDate = `\u2014`;
    if (timestamp) {
      let readableDate = new Date(timestamp);
      displayDate = readableDate.toLocaleString();
    }
    return displayDate;
  };

  const parseTimePeriod = (queryUrl: string) => {
    let [timeStringRegEx, fromDateString, toDateString] = queryUrl.match(
      timeRangeMatcher
    );

    fromDateString = fromDateString.replace(/[']+/g, '');
    toDateString = toDateString.replace(/[']+/g, '');

    let fromDateParsed = dateMath.parse(fromDateString);
    let toDateParsed = dateMath.parse(toDateString);

    const fromTimePeriod = fromDateParsed?.toDate();
    const toTimePeriod = toDateParsed?.toDate();
    return (
      fromTimePeriod?.toLocaleString() + ' -> ' + toTimePeriod?.toLocaleString()
    );
  };

  const getReportDetailsData = (report: ReportSchemaType) => {
    const {
      report_definition: reportDefinition,
      last_updated: lastUpdated,
      state,
      query_url: queryUrl,
    } = report;
    const { report_params: reportParams, trigger, delivery } = reportDefinition;
    const {
      trigger_type: triggerType,
      trigger_params: triggerParams,
    } = trigger;
    const {
      delivery_type: deliveryType,
      delivery_params: deliveryParams,
    } = delivery;
    const coreParams = reportParams.core_params;
    // covert timestamp to local date-time string
    let reportDetails = {
      reportName: reportParams.report_name,
      description:
        reportParams.description === '' ? `\u2014` : reportParams.description,
      created: convertTimestamp(report.time_created),
      lastUpdated: convertTimestamp(report.last_updated),
      source: reportParams.report_source,
      // TODO:  we have all data needed, time_from, time_to, time_duration,
      // think of a way to better display
      time_period: parseTimePeriod(queryUrl),
      defaultFileFormat: coreParams.report_format,
      state: state,
      reportHeader: reportParams.core_params.hasOwnProperty('header') && reportParams.core_params.header != ""
        ? converter.makeMarkdown(reportParams.core_params.header)
        : `\u2014`,
      reportFooter: reportParams.core_params.hasOwnProperty('footer') && reportParams.core_params.footer != ""
        ? converter.makeMarkdown(reportParams.core_params.footer)
        : `\u2014`,
      triggerType: triggerType,
      scheduleType: triggerParams ? triggerParams.schedule_type : `\u2014`,
      scheduleDetails: `\u2014`,
      alertDetails: `\u2014`,
      channel: deliveryType,
      emailRecipients:
        deliveryType === 'Channel' ? deliveryParams.recipients : `\u2014`,
      emailSubject:
        deliveryType === 'Channel' ? deliveryParams.title : `\u2014`,
      emailBody:
        deliveryType === 'Channel' ? deliveryParams.textDescription : `\u2014`,
      queryUrl: queryUrl,
    };
    return reportDetails;
  };

  useEffect(() => {
    const { httpClient } = props;
    httpClient
      .get('../api/reporting/reports/' + reportId)
      .then((response) => {
        handleReportDetails(getReportDetailsData(response));
        props.setBreadcrumbs([
          {
            text: 'Reporting',
            href: '#',
          },
          {
            text:
              'Report details: ' +
              response.report_definition.report_params.report_name,
          },
        ]);
      })
      .catch((error) => {
        console.log('Error when fetching report details: ', error);
        handleErrorToast();
      });
  }, []);

  const downloadIconDownload = async () => {
    handleLoading(true);
    await generateReportById(
      reportId,
      props.httpClient,
      handleSuccessToast,
      handleErrorToast,
      handlePermissionsMissingDownloadToast
    );
    handleLoading(false);
  };

  const fileFormatDownload = (data) => {
    let formatUpper = data['defaultFileFormat'];
    formatUpper = fileFormatsUpper[formatUpper];
    return (
      <EuiLink onClick={downloadIconDownload}>
        {formatUpper + ' '}
        <EuiIcon type="importAction" />
      </EuiLink>
    );
  };

  const sourceURL = (data) => {
    return (
      <EuiLink href={`${data.queryUrl}`} target="_blank">
        {data['source']}
      </EuiLink>
    );
  };

  const showLoadingModal = showLoading ?
    <GenerateReportLoadingModal setShowLoading={setShowLoading} /> : null; 

  return (
    <EuiPage>
      <EuiPageBody>
        <EuiTitle size="l">
          <h1>Report details</h1>
        </EuiTitle>
        <EuiSpacer size="m" />
        <EuiPageContent panelPaddingSize={'l'}>
          <EuiPageHeader>
            <EuiFlexGroup>
              <EuiFlexItem>
                <EuiPageHeaderSection>
                  <EuiTitle>
                    <h2>{reportDetails['reportName']}</h2>
                  </EuiTitle>
                </EuiPageHeaderSection>
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiPageHeader>
          <EuiHorizontalRule />
          <EuiTitle>
            <h3>Report Settings</h3>
          </EuiTitle>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Name'}
              reportDetailsComponentContent={reportDetails['reportName']}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Description'}
              reportDetailsComponentContent={reportDetails['description']}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Created'}
              reportDetailsComponentContent={reportDetails['created']}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Last updated'}
              reportDetailsComponentContent={reportDetails['lastUpdated']}
            />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Source'}
              reportDetailsComponentContent={sourceURL(reportDetails)}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Time period'}
              reportDetailsComponentContent={reportDetails.time_period}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'File format'}
              reportDetailsComponentContent={fileFormatDownload(reportDetails)}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'State'}
              reportDetailsComponentContent={reportDetails['state']}
            />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Report header'}
              reportDetailsComponentContent={trimAndRenderAsText(
                reportDetails['reportHeader']
              )}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Report footer'}
              reportDetailsComponentContent={trimAndRenderAsText(
                reportDetails['reportFooter']
              )}
            />
            <ReportDetailsComponent />
            <ReportDetailsComponent />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiTitle>
            <h3>Report trigger</h3>
          </EuiTitle>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Report type'}
              reportDetailsComponentContent={reportDetails['triggerType']}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Schedule type'}
              reportDetailsComponentContent={reportDetails['scheduleType']}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Schedule details'}
              reportDetailsComponentContent={reportDetails['scheduleDetails']}
            />
            <ReportDetailsComponent />
          </EuiFlexGroup>
          <EuiSpacer />
          {/* <EuiTitle>
            <h3>Notification settings</h3>
          </EuiTitle>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Email recipient(s)'}
              reportDetailsComponentContent={formatEmails(
                reportDetails['emailRecipients']
              )}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Email subject'}
              reportDetailsComponentContent={reportDetails['emailSubject']}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Optional message'}
              reportDetailsComponentContent={trimAndRenderAsText(
                reportDetails['emailBody']
              )}
            />
            <ReportDetailsComponent />
          </EuiFlexGroup> */}
        </EuiPageContent>
        <EuiGlobalToastList
          toasts={toasts}
          dismissToast={removeToast}
          toastLifeTimeMs={6000}
        />
        {showLoadingModal}
      </EuiPageBody>
    </EuiPage>
  );
}
