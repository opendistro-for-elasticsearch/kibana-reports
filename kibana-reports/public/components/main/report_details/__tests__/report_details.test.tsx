/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { ReportDetails } from '../report_details';
import propsMock from '../../../../../test/propsMock';
import httpClientMock from '../../../../../test/httpMockClient';
import 'babel-polyfill';
import { act } from 'react-dom/test-utils';

function setBreadcrumbs(array: []) {
  jest.fn();
}

describe('<ReportDetails /> panel', () => {
  const match = {
    params: {
      reportId: '1',
    },
  };

  test('render on-demand component', async () => {
    const promise = Promise.resolve();
    const report_definition = {
      report_params: {
        report_name: 'test create report definition trigger',
        report_source: 'Dashboard',
        description: '',
        core_params: {
          base_url: '',
          report_format: '',
          header: '',
          footer: '',
          time_duration: 'PT30M',
        },
      },
      delivery: {
        delivery_type: '',
        delivery_params: {},
      },
      trigger: {
        trigger_type: 'On demand',
      },
    };

    httpClientMock.get = jest.fn().mockResolvedValue({
      report_definition,
      query_url: `http://localhost:5601/app/dashboards#/view/7adfa750-4c81-11e8-b3d7-01146121b73d?_g=(time:(from:'2020-10-23T20:53:35.315Z',to:'2020-10-23T21:23:35.316Z'))`,
    });

    const { container } = render(
      <ReportDetails
        httpClient={httpClientMock}
        props={propsMock}
        match={match}
        setBreadcrumbs={setBreadcrumbs}
      />
    );
    await act(() => promise);
    await expect(container.firstChild).toMatchSnapshot();
  });

  test('render 5 hours recurring component', async () => {
    const promise = Promise.resolve();
    const report_definition = {
      report_params: {
        report_name: 'test create report definition trigger',
        report_source: 'Dashboard',
        description: '',
        core_params: {
          base_url: '',
          report_format: '',
          header: '',
          footer: '',
          time_duration: 'PT30M',
        },
      },
      delivery: {
        delivery_type: '',
        delivery_params: {},
      },
      trigger: {
        trigger_type: 'Schedule',
        trigger_params: {
          schedule_type: 'Recurring',
          schedule: {
            interval: {
              period: 5,
              unit: 'HOURS',
              timezone: 'PST8PDT',
            },
          },
          enabled_time: 1114939203,
          enabled: true,
        },
      },
    };

    httpClientMock.get = jest.fn().mockResolvedValue({
      report_definition,
      query_url: `http://localhost:5601/app/dashboards#/view/7adfa750-4c81-11e8-b3d7-01146121b73d?_g=(time:(from:'2020-10-23T20:53:35.315Z',to:'2020-10-23T21:23:35.316Z'))`,
    });

    const { container } = render(
      <ReportDetails
        httpClient={httpClientMock}
        props={propsMock}
        match={match}
        setBreadcrumbs={setBreadcrumbs}
      />
    );
    await act(() => promise);
    await expect(container.firstChild).toMatchSnapshot();
  });
});
