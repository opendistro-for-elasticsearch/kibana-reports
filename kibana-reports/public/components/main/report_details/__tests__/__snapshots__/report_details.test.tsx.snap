// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportDetails /> panel render 5 hours recurring component 1`] = `
<div
  class="euiPage"
>
  <main
    class="euiPageBody"
  >
    <h1
      class="euiTitle euiTitle--large"
    >
      Report details
    </h1>
    <div
      class="euiSpacer euiSpacer--m"
    />
    <div
      class="euiPanel euiPanel--paddingLarge euiPageContent"
    >
      <div
        class="euiPageHeader euiPageHeader--responsive"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <div
              class="euiPageHeaderSection"
            >
              <h2
                class="euiTitle euiTitle--medium"
              >
                test create report definition trigger
              </h2>
            </div>
          </div>
        </div>
      </div>
      <hr
        class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
      />
      <h3
        class="euiTitle euiTitle--medium"
      >
        Report Settings
      </h3>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Name
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              test create report definition trigger
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Description
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              —
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Created
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              —
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Last updated
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              —
            </dd>
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Source
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              <a
                class="euiLink euiLink--primary"
                href="http://localhost:5601/app/dashboards#/view/7adfa750-4c81-11e8-b3d7-01146121b73d?_g=(time:(from:'2020-10-23T20:53:35.315Z',to:'2020-10-23T21:23:35.316Z'))"
                rel="noopener noreferrer"
                target="_blank"
              >
                Dashboard
              </a>
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Time period
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              10/23/2020, 1:53:35 PM -&gt; 10/23/2020, 2:23:35 PM
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              File format
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              <button
                class="euiLink euiLink--primary"
                type="button"
              >
                undefined 
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 10.114l1.85-1.943a.52.52 0 01.77 0c.214.228.214.6 0 .829l-1.95 2.05a1.552 1.552 0 01-2.31 0L5.41 9a.617.617 0 010-.829.52.52 0 01.77 0L8 10.082V1.556C8 1.249 8.224 1 8.5 1s.5.249.5.556v8.558zM4.18 6a.993.993 0 00-.972.804l-1.189 6A.995.995 0 002.991 14h11.018a1 1 0 00.972-1.196l-1.19-6a.993.993 0 00-.97-.804H4.18zM6 5v1h5V5h1.825c.946 0 1.76.673 1.946 1.608l1.19 6A2 2 0 0114.016 15H2.984a1.992 1.992 0 01-1.945-2.392l1.19-6C2.414 5.673 3.229 5 4.174 5H6z"
                  />
                </svg>
              </button>
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              State
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Report header
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              <p>
                —
              </p>
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Report footer
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              <p>
                —
              </p>
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            />
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            />
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <h3
        class="euiTitle euiTitle--medium"
      >
        Report trigger
      </h3>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Report type
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              Schedule
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Schedule type
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              Recurring
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Schedule details
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              —
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            />
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
    </div>
    <div
      aria-live="polite"
      class="euiGlobalToastList euiGlobalToastList--right"
      role="region"
    />
  </main>
</div>
`;

exports[`<ReportDetails /> panel render on-demand component 1`] = `
<div
  class="euiPage"
>
  <main
    class="euiPageBody"
  >
    <h1
      class="euiTitle euiTitle--large"
    >
      Report details
    </h1>
    <div
      class="euiSpacer euiSpacer--m"
    />
    <div
      class="euiPanel euiPanel--paddingLarge euiPageContent"
    >
      <div
        class="euiPageHeader euiPageHeader--responsive"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <div
              class="euiPageHeaderSection"
            >
              <h2
                class="euiTitle euiTitle--medium"
              >
                test create report definition trigger
              </h2>
            </div>
          </div>
        </div>
      </div>
      <hr
        class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
      />
      <h3
        class="euiTitle euiTitle--medium"
      >
        Report Settings
      </h3>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Name
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              test create report definition trigger
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Description
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              —
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Created
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              —
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Last updated
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              —
            </dd>
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Source
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              <a
                class="euiLink euiLink--primary"
                href="http://localhost:5601/app/dashboards#/view/7adfa750-4c81-11e8-b3d7-01146121b73d?_g=(time:(from:'2020-10-23T20:53:35.315Z',to:'2020-10-23T21:23:35.316Z'))"
                rel="noopener noreferrer"
                target="_blank"
              >
                Dashboard
              </a>
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Time period
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              10/23/2020, 1:53:35 PM -&gt; 10/23/2020, 2:23:35 PM
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              File format
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              <button
                class="euiLink euiLink--primary"
                type="button"
              >
                undefined 
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </button>
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              State
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Report header
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              <p>
                —
              </p>
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Report footer
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              <p>
                —
              </p>
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            />
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            />
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <h3
        class="euiTitle euiTitle--medium"
      >
        Report trigger
      </h3>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Report type
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              On demand
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Schedule type
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              —
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Schedule details
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              —
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            />
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
    </div>
    <div
      aria-live="polite"
      class="euiGlobalToastList euiGlobalToastList--right"
      role="region"
    />
  </main>
</div>
`;
